R/utils_render_latex.R

```diff
@@ -376,138 +370,134 @@ create_body_rows <- function(n_rows,
+                type = "row"
+              ),
+              collapse = ""
+            )
 
-          } else {
-            summary_rows <- ""
-          }
-          summary_rows
+          paste0(summary_h_border, summary_rows)
         }
       )
     )
   )
 }
 
-create_grand_summary_rows <- function(n_cols,
-                                      boxh,
-                                      list_of_summaries,
-                                      stub_available) {
+create_grand_summary_rows_l <- function(list_of_summaries,
+                                        boxh) {
 
   if (
     length(list_of_summaries) < 1 ||
-    is.null(list_of_summaries$summary_df_display_list$`::GRAND_SUMMARY`)
+    is.null(list_of_summaries$summary_df_display_list$`::GRAND_SUMMARY`) ||
+    nrow(list_of_summaries$summary_df_display_list$`::GRAND_SUMMARY`) < 1
   ) {
     return("")
   }
 
-  default_vars <- dplyr::pull(dplyr::filter(boxh, type == "default"), "var")
+  default_vars <- boxh[boxh$type == "default", "var", drop = TRUE]
 
   grand_summary_df <-
     list_of_summaries$summary_df_display_list$`::GRAND_SUMMARY` %>%
-    dplyr::select(rowname, .env$default_vars)
-
-  body_content_grand_summary <- as.vector(t(grand_summary_df))
+    dplyr::select(.data$rowname, .env$default_vars)
 
   row_splits_summary <-
-    split_body_content(
-      body_content = body_content_grand_summary,
-      n_cols = n_cols
+    split_row_content(
+      grand_summary_df,
+      n_cols = ncol(grand_summary_df)
     )
 
-  if (length(row_splits_summary) > 0) {
-
-    grand_summary_rows <-
-      paste(
-        vapply(
-          row_splits_summary,
-          FUN.VALUE = character(1),
-          latex_body_row,
-          type = "row"
-        ),
-        collapse = ""
-      )
-
-    grand_summary_rows <- paste0(grand_summary_h_border, grand_summary_rows)
-
-  } else {
-    grand_summary_rows <- ""
-  }
+  grand_summary_rows <-
+    paste(
+      vapply(
+        row_splits_summary,
+        FUN.VALUE = character(1),
+        latex_body_row,
+        type = "row"
+      ),
+      collapse = ""
+    )
 
-  grand_summary_rows
+  paste0(grand_summary_h_border, grand_summary_rows)
 }
 
 # Define horizontal border line types for
 # summary rows and for grand summary rows
 summary_h_border <- "\\midrule \n"
 grand_summary_h_border <- "\\midrule \n\\midrule \n"
 
-#' Split the body content vector into a list structure
+#' Split data frame or matrix row content into a list structure
+#'
+#' This function takes any data frame or matrix and creates a list
+#' with every component representing a row, each containing a vector
+#' with length corresponding to the total number of columns in the
+#' finalized table
 #'
-#' Taking the `body_content` vector, split into list components with one item
-#' per row in the output table
 #' @noRd
-split_body_content <- function(body_content,
-                               n_cols) {
+split_row_content <- function(x, n_cols) {
```

Comment:
Get rid of `n_cols` argument, derive it from `ncol(x)` instead; and no need to return `rep("", ncols)`, just return `NULL`

Reply:
Thanks.